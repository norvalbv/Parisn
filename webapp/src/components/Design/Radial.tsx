import { FC, ReactElement } from "react";

interface RadialProps {
  size?: keyof typeof sizeMap;
  color?: string;
}

const Radial: FC<RadialProps> = ({
  size = "md",
  color = "primary",
}): ReactElement => {
  return (
    <div
          className={`absolute w-[30rem] h-[30rem] top-5 left-5 rounded-full bg-[#111] opacity-100 blur-3xl`}
    />
  );
};

export default Radial;

const sizeMap = {
  sm: "5",
  md: "10",
  lg: "15",
  xl: "20",
  "2xl": "25",
};
