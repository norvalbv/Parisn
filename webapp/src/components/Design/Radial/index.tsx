import React, { ReactElement } from 'react';
import { Colour } from 'types/tailwind';

export interface RadialProps {
  size?: keyof typeof sizeMap;
  colour?: Colour;
  classes?: string;
}

const Radial = ({
  size = 'xl',
  colour = 'darkpurple',
  classes = '',
}: RadialProps): ReactElement => {
  return (
    <svg
      className={`absolute rounded-full opacity-20 blur-3xl z-0 ${classes}`}
      style={{
        backgroundColor: colour,
        width: sizeMap[size],
        height: sizeMap[size],
      }}
    />
  );
};

export default Radial;

const sizeMap = {
  sm: '15rem',
  md: '25rem',
  lg: '35rem',
  xl: '40rem',
  '2xl': '45rem',
};
