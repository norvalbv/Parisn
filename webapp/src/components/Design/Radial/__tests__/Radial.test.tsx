import { FC } from 'react';
import renderer from 'react-test-renderer';

import Radial, { RadialProps } from '../';

const RadialWithRequiredProps: FC<Required<RadialProps>> = Radial;

describe('<Radial />', () => {
  test('Radial props with mandatory props', () => {
    const tree = renderer.create(<Radial />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Radial props with all props as required', () => {
    const tree = renderer
      .create(<RadialWithRequiredProps size="md" classes="bg-red-500" colour="blue" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
