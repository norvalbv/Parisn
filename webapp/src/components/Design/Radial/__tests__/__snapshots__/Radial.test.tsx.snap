// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Radial /> Radial props with all props as required 1`] = `
<svg
  className="absolute rounded-full opacity-20 blur-3xl z-0 bg-red-500"
  style={
    {
      "backgroundColor": "blue",
      "height": "25rem",
      "width": "25rem",
    }
  }
/>
`;

exports[`<Radial /> Radial props with mandatory props 1`] = `
<svg
  className="absolute rounded-full opacity-20 blur-3xl z-0 "
  style={
    {
      "backgroundColor": "darkpurple",
      "height": "40rem",
      "width": "40rem",
    }
  }
/>
`;
